src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -1934,3 +1960,324 @@ QGISEXTERN QgsVectorLayerImport::ImportError createEmptyLayer(
+  }
+  return execOk;
+}
+
+
+QGISEXTERN QString loadStyle( const QString& uri, QString& errCause )
+{
+  QString style = "";
+  QgsDataSourceURI dsUri( uri );
+  // connect to database
+  QSqlDatabase mDatabase = QgsMssqlProvider::GetDatabase( dsUri.service(), dsUri.host(), dsUri.database(), dsUri.username(), dsUri.password() );
+
+  if ( !QgsMssqlProvider::OpenDatabase( mDatabase ) )
+  {
+    QgsDebugMsg( "Error connecting to database" );
+    QgsDebugMsg( mDatabase.lastError().text() );
+    return "";
+  }
+
+  QSqlQuery query = QSqlQuery( mDatabase );
+  query.setForwardOnly( true );
+
+  QString selectQmlQuery = QString( "SELECT top 1 styleQML"
+                                    " FROM layer_styles"
+                                    " WHERE f_table_catalog=%1"
+                                    " AND f_table_schema=%2"
+                                    " AND f_table_name=%3"
+                                    " AND f_geometry_column=%4"
+                                    " ORDER BY useAsDefault desc" )
+                           .arg( QgsMssqlProvider::quotedValue( dsUri.database() ) )
+                           .arg( QgsMssqlProvider::quotedValue( dsUri.schema() ) )
+                           .arg( QgsMssqlProvider::quotedValue( dsUri.table() ) )
+                           .arg( QgsMssqlProvider::quotedValue( dsUri.geometryColumn() ) );
+
+   if ( !query.exec( selectQmlQuery ) )
+   {
+     errCause = query.lastError().text();
+     QgsDebugMsg( errCause );
+     QgsDebugMsg("Load of Style failed");
+     return "";
+   }
+   if ( query.isActive() && query.next())
+   {
+     QString style = query.value( 0 ).toString();
+     return style;
+   }
+}
+
+QGISEXTERN int listStyles( const QString &uri, QStringList &ids, QStringList &names,
+                           QStringList &descriptions, QString& errCause )
+{
+  QgsDataSourceURI dsUri( uri );
+  // connect to database
+  QSqlDatabase mDatabase = QgsMssqlProvider::GetDatabase( dsUri.service(), dsUri.host(), dsUri.database(), dsUri.username(), dsUri.password() );
+
+  if ( !QgsMssqlProvider::OpenDatabase( mDatabase ) )
+  {
+    QgsDebugMsg( "Error connecting to database" );
+    QgsDebugMsg( mDatabase.lastError().text() );
+    return -1;
+  }
+
+  QSqlQuery query = QSqlQuery( mDatabase );
+  query.setForwardOnly( true );
+
+  // check if layer_styles table already exist
+  if ( !query.exec( QString( "SELECT COUNT(*) FROM information_schema.tables WHERE table_name= N'layer_styles'" ) ) )
+  {
+    QString msg = query.lastError().text();
+    errCause = msg;
+    QgsDebugMsg( msg );
+    return -1;
+  }
+  if ( query.isActive() && query.next() && query.value( 0 ).toInt() == 0 )
+  {
+    errCause = QObject::tr( "No styles available on DB, or there is an error connecting to the database." );
+    QgsDebugMsg(errCause);
```

Comment:
Change these two lines to:

```
QgsDebugMsg( QObject::tr( "No styles available on DB, or there is an error connecting to the database." ) );
```

(again, it avoids parsing the string when not using a debug build)